### PR TITLE
zh: refine TLS docs; add a FAQ about task upgrade

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What is changed, added or deleted? (Required)

- update docs about TLS to fix issues reported in:
    - https://docs.google.com/document/d/11GD1r5qbDVWLSSiNIJYWvVvi-xxeBbVYKcLN_4-Xhe4/edit# (internal only)
    - https://internal.pingcap.net/jira/browse/TOOL-2171 (internal only)
- add an FQA about upgrading DM 1.0 task to an existing DM 2.0 cluster.

### Which DM version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v3.0 changes)
- [x] v2.0 (TiDB DM 2.0 versions)
- [ ] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts after cherry-picked

### CODE REVIEW

**Comment by csuzhangxc** [comment]
@lance6716 @CharLotteiu PTAL

File: zh/enable-tls.md [hunk 1]
```diff
@@ -87,26 +93,34 @@ DM-master、DM-worker 和 dmctl 都会在每次新建相互通讯的连接时重
...
+    > 请确保所有 DM-master 与 DM-worker 组件能通过指定路径读取到证书与密钥文件的内容。
 
     ```yaml
     target-database:
         security:
-            ssl-ca: "/path/to/client-ca.pem"
-            ssl-cert: "/path/to/client-cert.pem"
-            ssl-key: "/path/to/client-key.pem"
+            ssl-ca: "/path/to/tidb-ca.pem"
+            ssl-cert: "/path/to/titdb-client-cert.pem"
```
Comment (yikeke) [inline zh/enable-tls.md:124]:
```suggestion
            ssl-cert: "/path/to/tidb-client-cert.pem"
```

File: zh/faq.md [hunk 2]
```diff
@@ -110,3 +110,12 @@ DM 在最后 `rename ghost_table to origin table` 的步骤会把内存的 DDL 
 - 任务配置文件中的配置项 `target-database.max-allowed-packet`（详情参见 [DM 任务完整配置文件介绍](task-configuration-file-full.md)）
 
 设置为比默认 67108864 (64M) 更大的值。详见 [Loader 解决方案](https://docs.pingcap.com/zh/tidb/stable/loader-overview#解决方案)。
+
+## 2.0 集群运行 1.0 已有数据迁移任务时报错 `Error 1054: Unknown column 'binlog_gtid' in 'field list'`
+
+在 DM 2.0 中，为 checkpoint 等元信息表引入了更多的字段。如果直接使用 1.0 集群的任务配置文件在 2.0 中通过 `start-task` 从增量复制阶段继续运行，则会出现 `Error 1054: Unknown column 'binlog_gtid' in 'field list'` 错误。
```
Comment (yikeke) [inline zh/faq.md:116]:
```suggestion
在 DM 2.0 中，为 checkpoint 等元信息表引入了更多的字段。如果在 2.0 中，通过 `start-task` 直接使用 1.0 集群的任务配置文件从增量复制阶段继续运行，则会出现 `Error 1054: Unknown column 'binlog_gtid' in 'field list'` 错误。
```

File: zh/generate-self-signed-certificates.md [hunk 3]
```diff
@@ -112,39 +112,51 @@ CA 的作用是签发证书。实际情况中，请联系你的管理员签发
...
+    
+    - `IP`
+    - `DNS`
+    - `URI`
+
+    > **注意：**
+    >
+    > 如果要使用 `0.0.0.0` 等特殊 IP 用于连接通讯，也需要将其加入到 `alt_names` 中。
+
+4. 保存 `openssl.cnf` 文件后，生成证书请求文件（在这一步当提供输入 `Common Name (e.g. server FQDN or YOUR name) []:` 时，可以为该证书指定 Common Name (CN)，如 `dm`。其作用是让服务端验证接入的客户端的身份，各个组件默认不会开启验证，需要在配置文件中启用该功能才生效）：
```
Comment (yikeke) [inline zh/generate-self-signed-certificates.md:125]:
```suggestion
4. 保存 `openssl.cnf` 文件后，生成证书请求文件（在这一步中提供 `Common Name (e.g. server FQDN or YOUR name) []:` 输入时，可以为该证书指定 Common Name (CN)，如 `dm`。其作用是让服务端验证接入的客户端的身份，各个组件默认不会开启验证，需要在配置文件中启用该功能才生效）：
```

File: zh/manually-upgrade-dm-1.0-to-2.0.md [hunk 4]
```diff
@@ -107,6 +107,8 @@ from:
 
 [使用 TiUP](deploy-a-dm-cluster-using-tiup.md) 按所需要节点数部署新的 v2.0.x 集群。
 
+如果已有其他可用的 v2.0.x 集群，则可跳过此步。
```
Comment (yikeke) [inline zh/manually-upgrade-dm-1.0-to-2.0.md:110]:
```suggestion

> **注意：**
> 
> 如果已有其他可用的 v2.0.x 集群，可跳过此步。

[使用 TiUP](deploy-a-dm-cluster-using-tiup.md) 按所需要节点数部署新的 v2.0.x 集群。
```
建议放到前面去，比较符合用户接受有效信息的顺序

**Comment by yikeke** [comment]
Please request a tech review? @csuzhangxc

**Comment by csuzhangxc** [comment]
> Please request a tech review? @csuzhangxc

@lance6716 will do it.

File: zh/faq.md [hunk 5]
```diff
@@ -110,3 +110,12 @@ DM 在最后 `rename ghost_table to origin table` 的步骤会把内存的 DDL 
 - 任务配置文件中的配置项 `target-database.max-allowed-packet`（详情参见 [DM 任务完整配置文件介绍](task-configuration-file-full.md)）
 
 设置为比默认 67108864 (64M) 更大的值。详见 [Loader 解决方案](https://docs.pingcap.com/zh/tidb/stable/loader-overview#解决方案)。
+
+## 2.0 集群运行 1.0 已有数据迁移任务时报错 `Error 1054: Unknown column 'binlog_gtid' in 'field list'`
+
+在 DM 2.0 中，为 checkpoint 等元信息表引入了更多的字段。如果在 2.0 中，通过 `start-task` 直接使用 1.0 集群的任务配置文件从增量复制阶段继续运行，则会出现 `Error 1054: Unknown column 'binlog_gtid' in 'field list'` 错误。
+
+对于此错误，可使用以下方式之一进行处理：
```
Comment (CharLotteiu) [inline zh/faq.md:118]:
```suggestion
对于此错误，可使用以下任一方式进行处理：
```

File: zh/faq.md [hunk 6]
```diff
@@ -110,3 +110,12 @@ DM 在最后 `rename ghost_table to origin table` 的步骤会把内存的 DDL 
...
+
+- [使用 TiUP 导入 DM 1.0 集群到全新的 2.0 集群](maintain-dm-using-tiup.md#导入-dm-ansible-部署的-dm-10-集群并升级)。
```
Comment (CharLotteiu) [inline zh/faq.md:120]:
```suggestion
- [使用 TiUP 将 DM 1.0 集群导入到全新的 2.0 集群](maintain-dm-using-tiup.md#导入-dm-ansible-部署的-dm-10-集群并升级)。
```

File: zh/faq.md [hunk 7]
```diff
@@ -110,3 +110,12 @@ DM 在最后 `rename ghost_table to origin table` 的步骤会把内存的 DDL 
...
+- [手动将 DM 1.0 的数据迁移任务导入到 2.0 集群](manually-upgrade-dm-1.0-to-2.0.md)。
```
Comment (lance6716) [inline zh/faq.md:121]: 好像在“手动文档”中暂缺add column的操作

**Review by lance6716 — COMMENTED** [review]
rest LGTM

File: zh/faq.md (hunk 7 above)
Comment (csuzhangxc) [reply zh/faq.md:121]: 不加了吧，那样后续如果我们对其他 meta 表有更多的更新，还得加更多的手动步骤。直接让用户使用 checkpoint 去启一个新的 task 吧？

**Comment by lance6716** [comment]
LGTM

**Review by yikeke — APPROVED** [review]

**Pull request merged**